To: vim_dev@googlegroups.com
Subject: Patch 7.3.418
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.418
Problem:    When a user complete function returns -1 an error message is
	    given.
Solution:   When -2 is returned stop completion silently. (Yasuhiro Matsumoto)
Files:	    src/edit.c


*** ../vim-7.3.417/src/edit.c	2012-01-20 14:32:22.000000000 +0100
--- src/edit.c	2012-01-26 18:32:23.000000000 +0100
***************
*** 5189,5194 ****
--- 5189,5199 ----
  		return FAIL;
  	    }
  
+ 	    /* Return value -2 means the user complete function wants to
+ 	     * cancel the complete without an error. */
+ 	    if (col == -2)
+ 		return FAIL;
+ 
  	    /*
  	     * Reset extended parameters of completion, when start new
  	     * completion.
*** ../vim-7.3.417/src/version.c	2012-01-26 18:58:13.000000000 +0100
--- src/version.c	2012-01-26 18:42:14.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     418,
  /**/

-- 
If someone questions your market projections, simply point out that your
target market is "People who are nuts" and "People who will buy any damn
thing".  Nobody is going to tell you there aren't enough of those people
to go around.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
